lib/ansible/plugins/connection/ssh.py

```diff
@@ -342,10 +351,10 @@ def _handle_error(remaining_retries, command, return_tuple, no_log, host, displa
             raise AnsibleAuthenticationFailure(msg)
 
         # sshpass returns codes are 1-6. We handle 5 previously, so this catches other scenarios.
-        # No exception is raised, so the connection is retried - except when attempting to use
-        # sshpass_prompt with an sshpass that won't let us pass -P, in which case we fail loudly.
+        # Raise exception to prevent retries. The failures will not be resolved with a simple retry.
+        # Fail loudly when attempting to use sshpass_prompt with an sshpass that won't let us pass -P
         elif return_tuple[0] in [1, 2, 3, 4, 6]:
```

Comment:
If we are going to raise an error for codes 1-6, then this conditional can be combined with the previous.

Originally, this was implemented to _only_ fail for incorrect password (error code 5) to prevent account lockouts. But if we're going to raise an exception for any error 1-6, then we should deduplicate this code.

Reply:
@samdoran Thank you for the suggestion. I can combine `sshpass` error code 5 (Incorrect password) and other 1-4,6 codes, but that will lead to a big changes in a code base. In this case, I think, we need to fully remove a logic about `remaining_retries` in _handle_error(). As for me, it's useless there in this case. Please double check this part of the code and let me know if I can proceed.

Reply:
I think if these are combined and we are going to raise an exception on the first encounter of error codes 1-4, 6, then it's safe to remove the `remaining_retries` conditional.